src/build_options.rs

```diff
@@ -129,7 +129,11 @@ impl BuildOptions {
         let cargo_toml = CargoToml::from_path(&manifest_file)?;
         let manifest_dir = manifest_file.parent().unwrap();
         let pyproject: Option<PyProjectToml> = if manifest_dir.join("pyproject.toml").is_file() {
-            Some(PyProjectToml::new(manifest_dir).context("pyproject.toml is invalid")?)
+            let pyproject =
+                PyProjectToml::new(manifest_dir).context("pyproject.toml is invalid")?;
+            pyproject.warn_missing_maturin_version();
+            pyproject.warn_missing_build_backend();
```

Comment:
Personally I'd just make these two functions as private functions in `build_options.rs` rather than methods on `PyProjectToml`. I don't think the type actually needs to care about these fields missing?

However that's an extreme nitpick so ignore me if you prefer the current structure.